controllers/oauth2client_controller.go

```diff
@@ -457,3 +479,17 @@ func removeString(slice []string, s string) (result []string) {
 	}
 	return
 }
+
+func determineApiSecretName(spec *hydrav1alpha1.HydraAdmin) string {
+	if spec.ApiKeySecretRef.Name != "" {
+		return spec.ApiKeySecretRef.Name
```

Comment:
Since the name should be always given, this shouldn't be required, or throw an error if the name is empty